pkg/api/types.go

```diff
@@ -343,12 +344,56 @@ type Result struct {
 
 // BuildInfo holds information about a particular step in the build process.
 type BuildInfo struct {
+	// StepInfo holds information about build steps.
+	StepInfo []BuildStepInfo
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
 	// build, failed.
 	FailureReason FailureReason
 }
 
+// BuildStepInfo holds information about a build step.
+type BuildStepInfo struct {
+	// StepName is the name of the build step.
+	Name StepName
+
+	// StartTime is the start time of the build step in the build process.
+	StartTime time.Time
+	// StopTime is the time the build step has finished (Successfully or not) in
+	// the build process.
+	StopTime time.Time
+}
+
+// StepName is the name of a build step.
+type StepName string
+
+const (
+	// AssembleScriptStep is the build step when the Assemble script is being
+	// run.
+	AssembleScriptStep StepName = "AssembleScriptStep"
+
+	// GatherInputContentStep is the build step when we try to fetch the source for the
+	// build.
+	GatherInputContentStep StepName = "GatherInputContentStep"
```

Comment:
this is done by openshift outside of s2i at this point, so whatever number you are collecting here is probably not relevant/accurate.